io/eolearn/io/processing_api.py

```diff
@@ -0,0 +1,205 @@
+        return size_x, size_y
+
+    def execute(self, eopatch=None, bbox=None, time_interval=None):
+        ''' Make a WFS request to get valid dates, download an image for each valid date and store it in an EOPatch
+
+        :param eopatch: input EOPatch
+        :type eopatch: EOPatch
+        '''
+
+        if self.size is not None:
+            size_x, size_y = self.size
+        elif self.resolution is not None:
+            size_x, size_y = self.size_from_resolution(bbox, self.resolution)
+
+        responses = [shr.response('default', 'image/tiff'), shr.response('userdata', 'application/json')]
+        request = shr.body(
+            request_bounds=shr.bounds(crs=bbox.crs.opengis_string, bbox=list(bbox)),
+            request_data=[shr.data(data_type=self.data_source.api_identifier())],
+            request_output=shr.output(size_x=size_x, size_y=size_y, responses=responses),
+            evalscript=self.generate_evalscript()
+        )
+
+        request_args = dict(
+            url=SHConfig().get_sh_processing_api_url(),
+            headers={"accept": "application/tar", 'content-type': 'application/json'},
+            data_folder=self.cache_folder,
+            hash_save=bool(self.cache_folder),
+            request_type='POST',
+            data_type=MimeType.TAR
+        )
+
+        dates = self.get_dates(bbox, time_interval)
+        requests = (self.request_from_date(request, date, self.maxcc, self.time_difference) for date in dates)
+        requests = [DownloadRequest(post_values=payload, **request_args) for payload in requests]
+
+        LOGGER.debug('Downloading %d requests of type %s', len(requests), str(self.data_source))
+        LOGGER.debug('Downloading bands: [%s]', ', '.join(self.all_bands))
+        client = SentinelHubDownloadClient()
+        images = client.download(requests)
+        LOGGER.debug('Downloads complete')
+
+        images = ((img['default.tif'], img['userdata.json']) for img in images)
+        images = [(img, meta.get('norm_factor', 0) if meta else 0) for img, meta in images]
+
+        eopatch = EOPatch() if eopatch is None else eopatch
+
+        eopatch.timestamp = dates
+        eopatch.bbox = bbox
+
+        shape = len(dates), size_y, size_x
+
+        # extract additional_data from the received images each as a separate feature
+        for f_type, f_name_src, f_name_dst in FeatureParser(self.additional_data, new_names=True)():
+            idx = self.all_bands.index(f_name_src)
+            feature_arrays = [np.atleast_3d(img)[..., idx] for img, norm_factor in images]
+            eopatch[(f_type, f_name_dst)] = np.asarray(feature_arrays).reshape(*shape, 1).astype(np.bool)
+
+        # extract bands from the received and save them as self.bands_feature
+        if self.bands:
+            img_bands = len(self.bands)
+            img_arrays = [img[..., slice(img_bands)].astype(np.float32) * norm_factor for img, norm_factor in images]
```

Comment:
I think its clearer if you write `img[..., :img_bands]` instead of `img[..., slice(img_bands)]`.

Reply:
True.

Reply:
Done.